tests/widget/inline/widget-inline.stage-attribute.test.js

```diff
@@ -9,13 +9,34 @@ test('Inline widget with additional stage attribute testing', async ({ page }) =
 
   await page.getByLabel('Username').fill('demouser');
 
-  await page.getByLabel('Password', { exact: true }).fill('j56eKtae*1');
+  await page.getByLabel('Password', { exact: true }).fill('failvalidation');
   await page.getByLabel('Confirm password').fill('not a match');
   await page.keyboard.press('Tab'); // focuses off confirm password
 
+  await expect(page.getByText('Passwords do not match')).toBeVisible();
+
   // TODO: Figure out why regular clicks don't work: https://github.com/microsoft/playwright/issues/13576
   await page.getByLabel('I agree').dispatchEvent('click');
 
+  await page.getByLabel('Confirm password').fill('failvalidation');
+  await page.keyboard.press('Tab'); // focuses off confirm password
+
+  // Ensure error message is removed
+  await expect(page.getByText('Passwords do not match')).toBeHidden();
+
+  // Resubmit form
+  await clickButton('Next', '/authenticate');
+
+  // Ensure error message is removed
+  await expect(page.getByText('Passwords do not match')).toBeHidden();
+
+  await expect(page.getByLabel('Password', { exact: true })).toHaveValue('');
+  await expect(page.getByLabel('Confirm password')).toHaveValue('');
+
+  await page.getByLabel('Password', { exact: true }).fill('j56eKtae*1');
+  await page.getByLabel('Confirm password').fill('not a match');
+  await page.keyboard.press('Tab'); // focuses off confirm password
+
```

Comment:
Adding more testing to the confirm password around client and server validation failures, ensuring we correctly report even after successful client validation, but failed server validation.